modules/tinymce/src/themes/silver/test/ts/browser/editor/ToolbarPersistTest.ts

```diff
@@ -0,0 +1,83 @@
+import { Chain, Log, NamedChain, Pipeline, Assertions, ApproxStructure, Waiter } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { Editor as McEditor } from '@ephox/mcagar';
+import { SugarElement, SugarBody, Insert, Focus, Remove } from '@ephox/sugar';
+import Theme from 'tinymce/themes/silver/Theme';
+import Editor from 'tinymce/core/api/Editor';
+
+UnitTest.asynctest('Toolbar persist test', (success, failure) => {
+  Theme();
+
+  const cWaitForEditorVisibility = (message: string, visible: boolean) => Waiter.cTryUntil(message, Chain.op((editor: Editor) => {
```

Comment:
Could this be done with `UiFinder.cWaitForVisible` / `UiFinder.cWaitForInvisible` instead?